tests/testsuite/update.rs

```diff
@@ -2096,3 +2095,273 @@ fn update_breaking_specific_packages() {
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                mixed-compatibility = "1.0"
+            "#,
+        )
+        .file("foo/src/lib.rs", "")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                mixed-compatibility = "2.0"
+            "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("mixed-compatibility", "2.0.1").publish();
+
+    p.cargo("update -Zunstable-options --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr(
+            "\
+[UPDATING] `[..]` index
+[UPGRADING] mixed-compatibility ^1.0 -> ^2.0
+[LOCKING] 1 package to latest compatible version
+[ADDING] mixed-compatibility v2.0.1
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_mixed_renaming() {
+    Package::new("renamed-from", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [workspace]
+                members = ["foo", "bar"]
+            "#,
+        )
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                renamed-to = { package = "renamed-from", version = "1.0" }
+            "#,
+        )
+        .file("foo/src/lib.rs", "")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                renamed-from = "1.0"
+            "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("renamed-from", "2.0.0").publish();
+
+    p.cargo("update -Zunstable-options --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr(
+            "\
+[UPDATING] `[..]` index
+[UPGRADING] renamed-from ^1.0 -> ^2.0
+[LOCKING] 1 package to latest compatible version
+[ADDING] renamed-from v2.0.0
+",
+        )
+        .run();
+
+    let foo_manifest = p.read_file("foo/Cargo.toml");
+    let bar_manifest = p.read_file("bar/Cargo.toml");
+
+    assert_e2e().eq(
+        &foo_manifest,
+        str![[r#"
+
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                renamed-to = { package = "renamed-from", version = "2.0" }
```

Comment:
@epage Hmm, this is a bug, then, isn't it?

Reply:
Yes

Reply:
I'm making a FIXME comment. Let's fix this in a separate PR.

Reply:
Never mind, I added the fix here.